### PR TITLE
Add NFTOffer directory

### PR DESCRIPTION
**Pull request by DennisDawson**
Add description of NFT Offer directory to DirectoryNode page.

### CODE REVIEW

File: docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md [hunk 1]
```diff
@@ -56,19 +55,34 @@ There are two kinds of Directories:
...
+{
+    "Flags": 0,
+    "Indexes": [
+        "91E23C060E8D46CBE436382659CF9E31DE3AEB630927D1657D88D033E330F085"
+    ],
+    "LedgerEntryType": "DirectoryNode",
+    "Owner": "rGSte5MQe6smA8ep5rPtKnVJG3JoySzKgs",
+    "RootIndex": "8696AF0E701716A471303F26F6671AF146ACD1ED34597FF10A3874EF2631DCC3",
+    "index": "8696AF0E701716A471303F26F6671AF146ACD1ED34597FF10A3874EF2631DCC3"
+}
```
Comment (shawnxie999) [inline docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md:77]: I dont think this example is correct. Here is a correct example, the NFT offer directory always has a `NFTokenID` field
```

{
   "result": {
      "index": "CC45A27DAF06BFA45E8AFC92801AD06A06B7004DAD0F7022E439B3A2F6FA5B5A",
      "ledger_current_index": 310,
      "node": {
         "Flags": 2,
         "Indexes": [
            "83C81AC39F9771DDBCD66F6C225FC76EFC0971384EC6148BAFA5BD18019FC495"
         ],
         "LedgerEntryType": "DirectoryNode",
         "NFTokenID": "000800009988C43C563A7BB35AF34D642990CDF089F11B445EB3DCCD00000132",
         "RootIndex": "CC45A27DAF06BFA45E8AFC92801AD06A06B7004DAD0F7022E439B3A2F6FA5B5A",
         "index": "CC45A27DAF06BFA45E8AFC92801AD06A06B7004DAD0F7022E439B3A2F6FA5B5A"
      },
      "status": "success",
      "validated": false
   }
}
```

Comment (DennisDawson) [reply docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md:77]: Thanks, Shawn. Did you get that result sending a DirectoryNode request? I'm not sure why I didn't get a similar result.

Comment (shawnxie999) [reply docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md:77]: Yes i got it by requesting `ledger_entry`. I think you may have used the wrong DirectoryNode. Usually, when an offer is created, multiple DirectoryNode could be changed, I think you may have used the wrong one.

File: docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md [hunk 2]
```diff
@@ -56,19 +55,42 @@ There are two kinds of Directories:
...
-| `LedgerEntryType`   | String    | UInt16            | Yes       | The value `0x0064`, mapped to the string `DirectoryNode`, indicates that this object is part of a Directory. |
-| `Owner`             | String    | AccountID         | No        | (Owner Directories only) The address of the account that owns the objects in this directory. |
+| `Indexes`           | Array     | Vector256         | Yes       | The contents of this directory: an array of IDs of other objects. |
+| `IndexNext`         | Number    | UInt64            | No        | If this directory consists of multiple pages, this ID links to the next object in the chain, wrapping around at the end. |
+| `IndexPrevious`     | Number    | UInt64            | No        | If this directory consists of multiple pages, this ID links to the previous object in the chain, wrapping around at the beginning. |
+| `LedgerEntryType`   | String    | UInt16            | Yes       | The value `0x0064`, mapped to the string `DirectoryNode`, indicates that this object is part of a directory. |
+| `NFTokenID`         | String    | Hash25            | Yes       | ID of the NFT in a buy or sell offer. |
```
Comment (shawnxie999) [inline docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md]: This field should be "No" for Required column. Also might want to mention that (NFT Offer directories only)

File: docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md [hunk 3]
```diff
@@ -56,19 +55,42 @@ There are two kinds of Directories:
...
+| `Owner`             | String    | AccountID         | No        | (Owner and NFT Offer Directories only) The address of the account that owns the objects in this directory. |
```
Comment (shawnxie999) [inline docs/references/protocol/ledger-data/ledger-entry-types/directorynode.md]: I dont think Owner is assciated with NFT Offer directories, should be just (Owner Directories only)

**Review by shawnxie999 — APPROVED** [review]

**Pull request merged**